adapters/zmaticoo/zmaticoo.go

```diff
@@ -0,0 +1,149 @@
+package zmaticoo
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/prebid/openrtb/v19/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/errortypes"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+	"net/http"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+// Builder builds a new instance of the zmaticoo adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	return &adapter{
+		endpoint: config.Endpoint,
+	}, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	adapterRequest, errs := a.makeRequest(request)
+	if errs != nil {
+		return nil, errs
+	}
+	return []*adapters.RequestData{adapterRequest}, nil
+
+}
+
+func (a *adapter) makeRequest(request *openrtb2.BidRequest) (*adapters.RequestData, []error) {
+	errs := getZmaticooExt(request)
```

Comment:
Method is named as `getZmaticooExt`. But Ext are not returned.

```
func getZmaticooExt(request *openrtb2.BidRequest) []error {
	var extImpZmaticoo openrtb_ext.ExtImpZmaticoo
	var errs []error
	for _, imp := range request.Imp {
		var extBidder adapters.ExtImpBidder
		err := json.Unmarshal(imp.Ext, &extBidder)
		if err != nil {
			errs = append(errs, err)
			continue
		}
		err = json.Unmarshal(extBidder.Bidder, &extImpZmaticoo)
		if err != nil {
			errs = append(errs, err)
			continue
		}
		if extImpZmaticoo.ZoneId == "" || extImpZmaticoo.PubId == "" {
			errs = append(errs, fmt.Errorf("imp.ext.pubId or imp.ext.zoneId required"))
			continue
		}
	}
	return errs

}
```

Based on above code,`getZmaticooExt` can be renamed to `validateZmaticooExt`

Reply:
ok，i will rename this function @onkarvhanumante